src/main/java/org/swisspush/redisques/action/AddQueueItemAction.java

```diff
@@ -17,17 +17,18 @@
 
 public class AddQueueItemAction extends AbstractQueueAction {
 
-    public AddQueueItemAction(Vertx vertx, LuaScriptManager luaScriptManager, RedisAPI redisAPI, String address, String queuesKey, String queuesPrefix,
-                                     String consumersPrefix, String locksKey, List<QueueConfiguration> queueConfigurations,
-                                     QueueStatisticsCollector queueStatisticsCollector, Logger log) {
-        super(vertx, luaScriptManager, redisAPI, address, queuesKey, queuesPrefix, consumersPrefix, locksKey, queueConfigurations,
+    public AddQueueItemAction(Vertx vertx, LuaScriptManager luaScriptManager, RedisAPIProvider redisAPIProvider, String address, String queuesKey, String queuesPrefix,
+                              String consumersPrefix, String locksKey, List<QueueConfiguration> queueConfigurations,
+                              QueueStatisticsCollector queueStatisticsCollector, Logger log) {
+        super(vertx, luaScriptManager, redisAPIProvider, address, queuesKey, queuesPrefix, consumersPrefix, locksKey, queueConfigurations,
                 queueStatisticsCollector, log);
     }
 
     @Override
     public void execute(Message<JsonObject> event) {
         String key1 = queuesPrefix + event.body().getJsonObject(PAYLOAD).getString(QUEUENAME);
         String valueAddItem = event.body().getJsonObject(PAYLOAD).getString(BUFFER);
-        redisAPI.rpush(Arrays.asList(key1, valueAddItem), new AddQueueItemHandler(event));
+        redisAPIProvider.redisAPI().onSuccess(
+                redisAPI -> redisAPI.rpush(Arrays.asList(key1, valueAddItem), new AddQueueItemHandler(event)));
```

Comment:
In the RedisQues class during initialization we handle errors with
```java
redisAPIProvider.redisAPI().onComplete(event -> {
	if (event.succeeded()) {
		initialize();
		promise.complete();
	} else {
		promise.fail(event.cause());
	}
});
```
so Redisques will not start when Redis is not available. For the other cases (when Redis breaks afterwards), we will not get into an error because we return the _RedisAP_ instance.
```java
public Future<RedisAPI> redisAPI() {
	if (redisAPI == null) {
		return createRedisClient();
	} else {
		return Future.succeededFuture(redisAPI);
	}
}
```
The behaviour will be the same as before. We have the _RedisAPI_ instance but don't get results from Redis. This leads to _EventBus_ timeouts which are handled.